kore/src/Kore/Step/Step.hs

```diff
@@ -428,54 +478,31 @@ applyRule
     -- ^ Evaluates functions.
     -> BuiltinAndAxiomSimplifierMap
     -- ^ Map from symbol IDs to defined functions
-    -> UnificationProcedure
 
-    -> Pattern variable
+    -> Pattern (Target variable)
```

Comment:
I think that this does not need to be a `Pattern`, it could be just a `Predicate`.